bvm/ballerina-core/src/main/java/org/ballerinalang/util/codegen/PackageInfoReader.java

```diff
@@ -1704,13 +1704,18 @@ private PackageInfo getPackageInfo(String pkgPath) {
         PackageInfo pkgInfo = programFile.getPackageInfo(pkgPath);
 
         // if the package info is not available in the balx file, then it should be read from the balo
-        if (pkgInfo == null) {
+        if (pkgInfo != null) {
+            return pkgInfo;
+        }
+
+        try {
             PackageFileReader pkgFileReader = new PackageFileReader(this.programFile);
             pkgFileReader.readPackage(pkgPath);
-            pkgInfo = programFile.getPackageInfo(pkgPath);
+        } catch (IOException e) {
+            throw new BLangRuntimeException("error while reading package: " + pkgPath, e);
```

Comment:
do we need the "while" here? At https://github.com/ballerina-platform/ballerina-lang/pull/10120/files#diff-3d428a91487a0dfe643619a3bb2722b5R46 we say "ballerina: error reading program file"?

Reply:
+1. updated